Sources/SwiftDocC/Infrastructure/DocumentationContext.swift

```diff
@@ -138,14 +138,13 @@ public class DocumentationContext: DocumentationContextDataProviderDelegate {
     }
     
     /// The root module nodes of the Topic Graph.
-    public var rootModules: [ResolvedTopicReference] {
-        return topicGraph.nodes.values.compactMap { node in
-            guard node.kind == .module,
-                  !onlyHasSnippetRelatedChildren(for: node.reference) else {
-                return nil
-            }
-            return node.reference
-        }
+    ///
+    /// This property is initialized during the registration of a documentation bundle.
+    public private(set) var rootModules: [ResolvedTopicReference]!
```

Comment:
Good question, the reason it's an implicitly unwrapped optional is because it would be a programmer error to use this API before a bundle has been registered (as indicated by the doc comment). We would want the app to crash if it's called too early so that we resolve the issue in the code (it's like a precondition failure). Providing a default value of `[]` would be misleading (in fact, the previous implementation had the same issue), because it's unclear whether the documentation actually has no root modules, or whether they just haven't been discovered yet.